mongofiles/mongofiles.go

```diff
@@ -467,8 +495,33 @@ func (mf *MongoFiles) Run(displayHost bool) (output string, finalErr error) {
 	case Get, GetID:
 		err = mf.handleGet()
 
-	case Put, PutID:
-		err = mf.handlePut()
+		// If mongofiles --put ... is called, i.e. with multiple supporting
+		// arguments, then add gridFiles specified in mf.FileNameList
+		// (see TOOLS-2667). Otherwise, if mongofiles --put_id is called, then
+		// preserve existing behavior.
+	case Put:
+		for _, file := range mf.FileNameList {
```

Comment:
```suggestion
		for _, filename := range mf.FileNameList {
```
[nit] naming; "filename" is more clear than "file", though this is my opinion so you can choose to ignore this suggestion.

Reply:
Done!